### PR TITLE
[clang][AArch64] Add --print-supported-extensions support

### PR DESCRIPTION
**Pull request by DavidSpickett**
This follows the RISC-V work done in 4b40ced4e5ba10b841516b3970e7699ba8ded572.

This uses AArch64's target parser instead. We just list the names, without the "+" on them, which matches RISC-V's format.

```
$ ./bin/clang -target aarch64-linux-gnu --print-supported-extensions
clang version 18.0.0 (https://github.com/llvm/llvm-project.git 154da8aec20719c82235a6957aa6e461f5a5e030)
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: <...>
All available -march extensions for AArch64

        aes
        b16b16
        bf16
        brbe
        crc
        crypto
        cssc
        <...>
```

Since our extensions don't have versions in the same way there's just one column with the name in.

Any extension without a feature name (including the special "none") is not listed as those cannot be passed to -march, they're just for the backend. For example the MTE extension can be added with "+memtag" but MTE2 and MTE3 do not have feature names so they cannot be added to -march.

This does not attempt to tackle the fact that clang allows invalid combinations of AArch64 extensions, it simply lists the possible options. It's still up to the user to ask for something sensible.

Equally, this has no context of what CPU is being selected. Neither does the RISC-V option, the user has to be aware of that.

I've added a target parser test, and a high level clang test that checks RISC-V and AArch64 work and that Intel, that doesn't support this, shows the correct error.

### CODE REVIEW

**Comment by DavidSpickett** [comment]
I've generalised this a bit but mostly kept to what RISC-V did. If we think it's time to completely generalise it then I'll figure that out.

Mostly want to know people are on board with supporting this for AArch64 first.

File: llvm/lib/TargetParser/AArch64TargetParser.cpp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 //===----------------------------------------------------------------------===//
 
 #include "llvm/TargetParser/AArch64TargetParser.h"
+#include "llvm/Support/raw_ostream.h"
```
Comment (DanielKristofKiss) [inline llvm/lib/TargetParser/AArch64TargetParser.cpp:15]: Did you run clang-format? IIRC this would be in alphabet order.

**Review by DanielKristofKiss — COMMENTED** [review]
just a NIT, LGTM otherwise.

File: llvm/lib/TargetParser/AArch64TargetParser.cpp (hunk 1 above)
Comment (DavidSpickett) [reply llvm/lib/TargetParser/AArch64TargetParser.cpp:15]: clang-format is putting the "main module" header first, then sorting alphabetically following: https://llvm.org/docs/CodingStandards.html#include-style

Comment (DanielKristofKiss) [reply llvm/lib/TargetParser/AArch64TargetParser.cpp:15]: mybad missed first is the main module.

**Review by DanielKristofKiss — APPROVED** [review]

**Pull request merged**